app_utils/sections/experiment.py

```diff
@@ -1758,95 +1757,16 @@ async def experiment_push_to_huggingface_dialog(q: Q, error: str = ""):
         user_id = q.client["experiment/display/push_to_huggingface/account_name"]
-        if user_id == "":
-            user_id = huggingface_hub.whoami()["name"]
-        exp_name = q.client[
+        safe_serialization = q.client["default_safe_serialization"]
+        model_name = q.client[
             "experiment/display/push_to_huggingface/model_name"
         ].replace(".", "-")
-        repo_id = f"{user_id}/{exp_name}"
 
-        # push tokenizer to hub
-        tokenizer.push_to_hub(repo_id=repo_id, private=True)
-
-        # push model card to hub
-        card = get_model_card(cfg, model, repo_id)
-        card.push_to_hub(
-            repo_id=repo_id, repo_type="model", commit_message="Upload model card"
-        )
-
-        # push config to hub
-        api = huggingface_hub.HfApi()
-        api.upload_file(
-            path_or_fileobj=f"{experiment_path}/cfg.yaml",
-            path_in_repo="cfg.yaml",
-            repo_id=repo_id,
-            repo_type="model",
-            commit_message="Upload cfg.yaml",
-        )
-
-        # push model to hub
-        model.backbone.config.custom_pipelines = {
-            "text-generation": {
-                "impl": "h2oai_pipeline.H2OTextGenerationPipeline",
-                "pt": "AutoModelForCausalLM",
-            }
-        }
-        model.backbone.push_to_hub(
-            repo_id=repo_id,
-            private=True,
-            commit_message="Upload model",
-            safe_serialization=q.client["default_safe_serialization"],
-        )
-
-        # Storing HF attributes
-        save_hf_yaml(
-            path=f"{cfg.output_directory}/hf.yaml",
-            account_name=user_id,
-            model_name=exp_name,
-            repo_id=repo_id,
-        )
-
-        # push pipeline to hub
-        template_env = Environment(
-            loader=FileSystemLoader(searchpath="llm_studio/src/")
-        )
-        pipeline_template = template_env.get_template("h2oai_pipeline_template.py")
-
-        data = {
-            "text_prompt_start": cfg.dataset.text_prompt_start,
-            "text_answer_separator": cfg.dataset.text_answer_separator,
-        }
-        if cfg.dataset.add_eos_token_to_prompt:
-            data.update({"end_of_sentence": cfg._tokenizer_eos_token})
-        else:
-            data.update({"end_of_sentence": ""})
-
-        custom_pipeline = pipeline_template.render(data)
-
-        custom_pipeline_path = os.path.join(experiment_path, "h2oai_pipeline.py")
-        with open(custom_pipeline_path, "w") as f:
-            f.write(custom_pipeline)
-
-        api.upload_file(
-            path_or_fileobj=custom_pipeline_path,
-            path_in_repo="h2oai_pipeline.py",
-            repo_id=repo_id,
-            repo_type="model",
-            commit_message="Upload h2oai_pipeline.py",
+        publish_model_to_hugging_face(
+            experiment_path, device, api_key, user_id, model_name, safe_serialization
```

Comment:
```suggestion
            path_to_experiment=experiment_path,
            model_name=model_name,
            user_id=user_id,
            api_key=api_key,
            device=device,
            safe_serialization=safe_serialization,
```

Reply:
@pascal-pfeiffer 

Done. Changed from positional arguments to keyword arguments.